// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

syntax = "proto2";

message t_dl_stats_item {
   required uint32 count = 1;
   required uint32 total = 2;
   required uint32 min = 3;
   required uint32 max = 4;
}

message t_dl_traffic_stats_item {
   required uint32 count = 1;
   required uint32 total = 2;
}

message t_dl_stats_am {

   // PDUs re-transmitted
   required t_dl_traffic_stats_item pdu_retx_bytes = 1;
   
   // STATUS PDUs retransmitted
   required t_dl_traffic_stats_item pdu_status_bytes = 2;

   // retx counts
   required t_dl_stats_item pdu_retx_count = 3;

   // pdu window 
   required t_dl_stats_item pdu_window = 4;
}

message t_dl_stats {
   required uint32 du_ue_index = 1;
   required uint32 is_srb = 2;
   required uint32 rb_id = 3;
   required uint32 rlc_mode = 4;

   required t_dl_stats_item sdu_queue_pkts = 5;
   required t_dl_stats_item sdu_queue_bytes = 6;

   required t_dl_traffic_stats_item sdu_new_bytes = 7;

   // PDUs transmitted
   required t_dl_traffic_stats_item pdu_tx_bytes = 8;
   
   // sdu_tx_latencies
   required t_dl_stats_item sdu_tx_started = 9;
   required t_dl_stats_item sdu_tx_completed = 10;
   required t_dl_stats_item sdu_tx_delivered = 11;

   optional t_dl_stats_am am = 12;  
}

message rlc_dl_stats {
   required uint64 timestamp = 1;
   repeated t_dl_stats stats = 2;
}




